webhooks/virtualmachine/mutation/virtualmachine_mutator_unit_test.go

```diff
@@ -272,6 +273,47 @@ func unitTestsMutating() {
 		})
 	})
 
+	Describe("SetDefaultBiosUUID", func() {
+
+		Context("When VM BiosUUID is empty", func() {
+			BeforeEach(func() {
+				ctx.vm.Spec.BiosUUID = ""
+			})
+
+			It("Should set BiosUUID", func() {
+				Expect(mutation.SetDefaultBiosUUID(&ctx.WebhookRequestContext, ctx.Client, ctx.vm)).To(BeTrue())
+				Expect(ctx.vm.Spec.BiosUUID).ToNot(BeEmpty())
+			})
+
+		})
+
+		Context("When VM BiosUUID is not empty", func() {
+			BeforeEach(func() {
+				ctx.vm.Spec.BiosUUID = uuid.New().String()
+			})
+
+			It("Should not mutate BiosUUID", func() {
+				oldVM := ctx.vm.DeepCopy()
+				Expect(mutation.SetDefaultBiosUUID(&ctx.WebhookRequestContext, ctx.Client, ctx.vm)).To(BeFalse())
+				Expect(ctx.vm.Spec.BiosUUID).Should(Equal(oldVM.Spec.BiosUUID))
+			})
+		})
+
+		Context("When CloudInit is not empty", func() {
+			BeforeEach(func() {
+				ctx.vm.Spec.BiosUUID = uuid.New().String()
+				ctx.vm.Spec.Bootstrap = &vmopv1.VirtualMachineBootstrapSpec{
+					CloudInit: &vmopv1.VirtualMachineBootstrapCloudInitSpec{},
```

Comment:
can we add another test where ctx.vm.Spec.BiosUUID and  vm.Spec.Bootstrap.CloudInit.InstanceId is preset?.

Reply:
Added